main.go

```diff
@@ -144,7 +145,32 @@ func main() {
 		level.Error(logger).Log("err", err)
 		os.Exit(1)
 	}
-	go interruptHandler(l, logger)
+
+	quitCh := make(chan struct{})
+	quitHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "Requesting termination... Goodbye!")
+		close(quitCh)
+	})
+
+	forbiddenAPINotEnabled := http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusForbidden)
+		w.Write([]byte("Lifecycle APIs are not enabled"))
+	})
+
+	if *enableLifeCycle {
+		r.Handler("PUT", *routePrefix+"/-/quit", quitHandler)
+		r.Handler("POST", *routePrefix+"/-/quit", quitHandler)
+	} else {
+		r.Handler("PUT", *routePrefix+"/-/quit", forbiddenAPINotEnabled)
+		r.Handler("POST", *routePrefix+"/-/quit", forbiddenAPINotEnabled)
+	}
+
+	r.Handler("GET", "/-/quit", http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		w.Write([]byte("Only POST or PUT requests allowed"))
```

Comment:
Nit: End sentence with a period.

Reply:
Yes, I will fix the message [here](https://github.com/prometheus/prometheus/blob/27bbc235e08ba00161909fa19a47e6420d6d99e7/web/web.go#L358) too